### PR TITLE
feat: Render field for JSON invocation params

### LINKED ISSUE #5252: [playground][ui] plumb through json invocation parameters
use the json invocation parameter type from the api and plumb it through to our form, use a json code block for editing

### PR DESCRIPTION
**Pull request by cephalization**
Future proof the InvocationParamsForm with support for future json invocation params, so that they can be rendered automatically should any be defined on the backend.

> [!IMPORTANT]
> Fields displayed in the video are just for demonstration, and will not appear in production. We do not currently ship any JSON invocation parameters that are not handled elsewhere in the UI.

https://github.com/user-attachments/assets/233ad520-71d7-4d09-980d-3078ae335bdd

Resolves #5252

### CODE REVIEW

File: app/src/components/code/JSONEditor.tsx [hunk 1]
```diff
@@ -24,24 +24,35 @@ export type JSONEditorProps = Omit<
...
   // Force a refresh of the editor when the jsonSchema changes
   // Code mirror does not automatically refresh when the extensions change
   const [schemaKey, setSchemaKey] = React.useState(0);
   useEffect(() => {
     setSchemaKey((prev) => prev + 1);
   }, [jsonSchema]);
+  const contentLength = props.value?.length;
+  // When optionalLint is false, this value will always be true
+  // This prevents the extensions useMemo from needlessly re-running when optionalLint is false
+  const hasContent = !optionalLint || (contentLength && contentLength > 0);
```
Comment (Parker-Stafford) [inline app/src/components/code/JSONEditor.tsx:46]: can we change this to be something like shouldLint? and then just use it below as well?

After reading a few times i'm kinda starting to grok this but it feels confusing, why did you need to add this? I think hinging the extensions based on the content is going to be rough

File: app/src/pages/playground/InvocationParameterJsonEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export function InvocationParameterJsonEditor({
+  initialValue,
+  onChange: handleChange,
+  label,
+}: {
+  initialValue: unknown;
+  // jsonValue is typed as any on the gql side to allow any json value to be set
+  // we can mostly assume this any is safe because we are safely parsing and stringifying incoming and outgoing json
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onChange: (value: any) => void;
```
Comment (Parker-Stafford) [inline app/src/pages/playground/InvocationParameterJsonEditor.tsx:21]: can we change to unknown?

File: app/src/pages/playground/InvocationParameterJsonEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    [handleChange]
+  );
+
+  return (
+    <div
+      css={css`
+        & .ac-view {
+          width: 100%;
+          z-index: 1000;
+        }
```
Comment (Parker-Stafford) [inline app/src/pages/playground/InvocationParameterJsonEditor.tsx:46]: was the view rendering below the dialog or something? that feels strange to me we have plenty of forms in modals / slideovers i think

File: app/src/utils/jsonUtils.ts [hunk 4]
```diff
@@ -40,3 +40,13 @@ export function safelyParseJSON(str: string) {
...
   }
 }
+
+export function safelyStringifyJSON(
+  ...args: Parameters<typeof JSON.stringify>
+) {
+  try {
+    return JSON.stringify(...args);
+  } catch (e) {
+    return null;
```
Comment (Parker-Stafford) [inline app/src/utils/jsonUtils.ts:50]: super small nit:
could match the signature of safely parse for consistency / flexibility

**Review by Parker-Stafford — APPROVED** [review]

File: app/src/components/code/JSONEditor.tsx (hunk 1 above)
Comment (cephalization) [reply app/src/components/code/JSONEditor.tsx:46]: The rename makes sense. I introduced this because otherwise, empty JSONEditor fields trigger a lint error.

Empty JSONEditor fields are valid in the invocation params form so we should only enable linting when there is content in that case.

The optionalLint prop was added so that this new functionality only takes place when optionalLint is true, otherwise it will keep linting at all times just like the original behavior.

For what it is worth, we only register extensions when content length goes from 0 -> n, or n -> 0, not on every keystroke.

File: app/src/pages/playground/InvocationParameterJsonEditor.tsx (hunk 3 above)
Comment (cephalization) [reply app/src/pages/playground/InvocationParameterJsonEditor.tsx:46]: Oops, meant to get rid of that, good catch. I was trying to fix clipped tooltips from codemirror when the editor is only 1 line tall.

File: app/src/components/code/JSONEditor.tsx (hunk 1 above)
Comment (cephalization) [reply app/src/components/code/JSONEditor.tsx]: Renamed and simplified logic

Comment (Parker-Stafford) [reply app/src/components/code/JSONEditor.tsx]: gotcha, that makes sense, i think controlling when the useMemo runs is probably good the names were just a bit confusing update looks good
